beta-src/src/models/OrderClass.ts

```diff
@@ -0,0 +1,149 @@
+import BoardClass from "./BoardClass";
+import TerritoryClass from "./TerritoryClass";
+import { IOrderData } from "./Interfaces";
+import UnitClass from "./UnitClass";
+
+export default class OrderClass {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(
+    public board: BoardClass,
+    public orderData: IOrderData,
+    public unit: UnitClass,
+  ) {}
```

Comment:
Why not just use a non-null assertion operator here as well?

Reply:
Can param properties be used with non-null assertion?

Reply:
param props? You're doing it in another file and you're not changing anything here in the constructor. You're using non-null assertion operators in the `TerritoryClass`

Reply:
ahhh I see what you meant. If I go with that, declarations and assignment will be separated. Steve suggested param properties to make code shorter.